types/underscore/index.d.ts

```diff
@@ -360,131 +360,71 @@ declare module _ {
+         * truth test. Short-circuits and stops traversing `collection` if a false
+         * element is found.
+         * @param collection The collection to evaluate.
+         * @param iteratee The truth test to apply.
+         * @param context `this` object in `iteratee`, optional.
+         * @returns True if all elements pass the truth test, otherwise false.
+         **/
+        every<V extends Collection<any>>(
+            collection: V,
+            iteratee?: Iteratee<V, boolean>,
+            context?: any
+        ): boolean;
 
         /**
-        * Returns true if the value is present in the list. Uses indexOf internally,
-        * if list is an Array.
-        * @param list Checks each element to see if `value` is present.
-        * @param value The value to check for within `list`.
-        * @return True if `value` is present in `list`, otherwise false.
-        **/
-        contains<T>(
-            list: _.List<T>,
-            value: T,
-            fromIndex?: number): boolean;
+         * @see every
+         **/
+        all: UnderscoreStatic['every'];
 
         /**
-        * @see _.contains
-        **/
-        contains<T>(
-            object: _.Dictionary<T>,
-            value: T): boolean;
+         * Returns true if any of the values in `collection` pass the `iteratee`
+         * truth test. Short-circuits and stops traversing `collection` if a
+         * true element is found.
+         * @param collection The collection to evaluate.
+         * @param iteratee The truth test to apply.
+         * @param context `this` object in `iteratee`, optional.
+         * @returns True if any element passed the truth test, otherwise false.
+         **/
+        some<V extends Collection<any>>(
+            collection: V,
+            iteratee?: Iteratee<V, boolean>,
+            context?: any
+        ): boolean;
 
         /**
-        * @see _.contains
-        **/
-        include<T>(
-            list: _.Collection<T>,
-            value: T,
-            fromIndex?: number): boolean;
+         * @see some
+         **/
+        any: UnderscoreStatic['some'];
 
         /**
-        * @see _.contains
-        **/
-        include<T>(
-            object: _.Dictionary<T>,
-            value: T): boolean;
+         * Returns true if the value is present in `collection`. Uses indexOf
+         * internally, if `collection` is a List. Use `fromIndex` to start your
+         * search at a given index.
+         * @param collection The collection to check for `value`.
+         * @param value The value to check `collection` for.
+         * @param fromIndex The index to start searching from, optional,
+         * default = 0, only used when `collection` is a List.
+         * @returns True if `value` is present in `collection` after
+         * `fromIndex`, otherwise false.
+         **/
+        contains<V extends Collection<any>>(
+            collection: V,
+            value: TypeOfCollection<V>,
```

Comment:
From my company's solution, I actually get a new error with a collection that is a `List<string>` and a value that is a `string | undefined` where TS used to infer `string | undefined` from `value` which was compatible enough with a `List<T>` where `T` is string that it went with it, but now it doesn't do that anymore. Overall I'm not opposed since a simple `value && _.contains(list, value)` will fix that and is cheaper than searching the list for an `undefined` that won't be in there, but if folks are opposed to this let me know and I can add a generic for `T` back to bring back the prior behavior.

Reply:
It's worth noting that this may also apply to other type unions that are more interesting than `string | undefined`.

Reply:
Sounds good, I'll try to find a middle ground that constrains the value so folks can't pass in something completely outrageous while allowing values that only have a chance of matching.

As a point of interest, I was curious about whether this worked this way in `master` for `Underscore` and `_Chain` since it constrains `value` to the class generic `T` rather than a function generic like `UnderscoreStatic` and found that this error was already happening for those interfaces in existing definitions.

Reply:
Regrettably `TValue extends T` didn't work like I was hoping so I went with the `any` approach since I figured I could always revisit this later if I came up with something better.